stdlib/public/TensorFlow/Gradients.swift

```diff
@@ -206,9 +206,8 @@ extension Tensor where Scalar : Differentiable & FloatingPoint {
   static func _vjpAdd(
     lhs: Tensor, rhs: Tensor
   ) -> (Tensor, (Tensor) -> (Tensor, Tensor)) {
-    let value = lhs + rhs
-    return (value, { v in
-      return (v.unbroadcast(like: lhs), v.unbroadcast(like: rhs))
+    return (lhs + rhs, { [rhsShape = rhs.shapeTensor] v in
+      return (v.unbroadcast(like: lhs), v.unbroadcast(toShape: rhsShape))
```

Comment:
If it's more efficient to store the shape of `rhs` than the whole tensor, doesn't the same logic apply for `lhs`?